python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1590,11 +1590,17 @@ def tuples_to_time_windows(
 
     @classmethod
     def _deserialize(
-        cls, serialized: str, partitions_def: TimeWindowPartitionsDefinition
+        cls, serialized: str, partitions_def: Optional[TimeWindowPartitionsDefinition]
     ) -> Tuple[TimeWindowPartitionsDefinition, Sequence[TimeWindow], int]:
         loaded = json.loads(serialized)
 
         if isinstance(loaded, list):
+            if partitions_def is None:
```

Comment:
This check error and the one below shouldn't be hit -- this would only happen for really old asset backfills where `can_deserialize` isn't called previously